packages/portfolio-contract/src/portfolio.exo.ts

```diff
@@ -266,17 +269,18 @@ export const preparePortfolioKit = (
           if (!extra.memo) return;
           const memo: AxelarGmpIncomingMemo = JSON.parse(extra.memo); // XXX unsound! use typed pattern
 
+          const result = Object.entries(axelarIds).find(
+            ([_, chainId]) => chainId === memo.source_chain,
+          );
```

Comment:
Rather than the `chainName as AxelarChain` cast later on, we get more help from `tsc` if we get the types right earlier:

```suggestion
          // cast to work around shortcoming in type of Object.entries
          const result = (Object.entries(axelarIds) as [AxelarChain, string][]).find(
            ([_, chainId]) => chainId === memo.source_chain,
          );
```